bundles/org.openhab.binding.mqtt.generic/src/main/resources/ESH-INF/config/trigger-channel-config.xml

```diff
@@ -0,0 +1,31 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<config-description:config-descriptions
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:config-description="https://openhab.org/schemas/config-description/v1.0.0"
+	xsi:schemaLocation="https://openhab.org/schemas/config-description/v1.0.0 https://openhab.org/schemas/config-description-1.0.0.xsd">
+
+	<config-description uri="thing-type:mqtt:trigger_channel">
+		<parameter-group name="transformations">
+			<label>Transform Values</label>
+			<description>These configuration parameters allow you to alter a value before it is published to MQTT or before a
+				received value is used in the trigger.</description>
+			<advanced>true</advanced>
+		</parameter-group>
+
+		<parameter name="stateTopic" type="text" required="true">
+			<label>MQTT Trigger Topic</label>
+			<description>An MQTT topic that this thing will subscribe to, to receive the trigger</description>
+		</parameter>
+		<parameter name="transformationPattern" type="text" groupName="transformations">
```

Comment:
A transformation doesn't make sense for a trigger

Reply:
Why not? 
The incoming transformations are apply before trigger is executed.
So one might want to map the incomming values of filter them using REGEX.

Reply:
Triggers don't have a value, there should only be a discrete set of trigger events. Using JSON and transforming it both seems very wrong here.

Reply:
Well, the idea would be, that the device sends different values for the different events of the trigger. 
And then it might very well be good to be able to MAP the values sent by the device to more 'standard' ones used be your rules.

Reply:
I tend to agree with @davidgraeff and @kaikreuzer. The idea of the payload is IMO is not to transport information recevived from the device but to present pre-defined values to the receiver.

E.g. an alarm system could send `ARMED` or `DISARMED` as payload, so a notification can be send. for MQTT it would be more "a value has been received". If you are interested in the value itself, a state channel with `changed` in the rules should be used.

Reply:
Just so I understand you right:
If a device sends the value `fired` on different topics (e.g. `../trigger/armed` or `../trigger/disarmed`) then it would be ok to map it to an OH trigger.
If the device sends different values (e.g. `armed`, `disarmed`) on a single topic `../trigger` then it is not ok to handle this as an OH trigger.

That does not resonate with me.
And how is reaching out to zigbee and convinces them that they are doing it in a bad way ;-)

Reply:
I think the difference is: are these values pre-defined by the binding (like `armed`, `disarmed` in the case of the alarm system) or arbitrary values (like a temperature)?

Reply:
Yes, but this here would be the place to do the pre-defining, right?
Then we should require a list of text values, which define the trigger options.
And we would need to check for any of the defined values before we activate the trigger.

BTW, this is also not done for the triggers created currently by homie. There we use just any property, which is sent as not retained to create a trigger.

Reply:
Soyou say the transformation should be used to reduce the number of values (e.g. 0-10 = LOW, 11-15 = OK, 16-= HIGH)? That makes sense. @kaikreuzer, WDYT?

Reply:
I could also be used to map different devices to the same events.
Like one device sends the string 'clicked' while the other sends 'pushed'.
The rules listening for the triggers could then be independent of the these differences.
I do not think about numbers here, but strings.

Reply:
Both examples like `armed`/`disarmed` and ranges of temperatures are actually states, so what isn't clear to me is why this isn't a state channel then? Triggers are only meant for the situations, where something like a button press occurs, which does not have any state. Triggers must not be misused for state change events.

Reply:
Please help me understand this!
    /**
     * System wide trigger {@link ChannelType} which triggers "SHORT_PRESSED", "DOUBLE_PRESSED" and "LONG_PRESSED"
     * events.
     */
    public static final ChannelType SYSTEM_BUTTON = ChannelTypeBuilder
            .trigger(new ChannelTypeUID(BINDING_ID, "button"), "Button")
            .withEventDescription(
                    new EventDescription(Arrays.asList(new EventOption(CommonTriggerEvents.SHORT_PRESSED, null),
                            new EventOption(CommonTriggerEvents.DOUBLE_PRESSED, null),
                            new EventOption(CommonTriggerEvents.LONG_PRESSED, null))))
            .build();

A trigger take event options.
There is a trigger for the butten and the events are 'pressed', 'double pressed', and 'long pressed'.
So it does make sense for the mqtt trigger to use the value sent by the device as the event option, right?
Now, why shouldn't we be able to map the values sent by differnt devices to common values?
Suppose device one sends '0' for a click, '2' for a double click and '3' for a long press.
Suppose another device sending 'click' for a click and 'hold' for a long press.
With a transformation we are able to map the event options of the different devices to common values, Then all the downstream components can handle the triggers without knowning different devices.

Reply:
Yes, if we are talking about push buttons that are integrated via MQTT, triggers are fine. What I was saying above was that all arguments were done with examples that do NOT fit to a trigger. Likewise the [example given in the code](https://github.com/openhab/openhab-addons/pull/6902/files#diff-7056907cc9e6d7e8229964567a02905aR23-R24), which would clearly be better modelled as a state channel instead of a trigger.
Furthermore, [this parameter](https://github.com/openhab/openhab-addons/pull/6902/files#diff-7056907cc9e6d7e8229964567a02905aR15) has "state" in its name, which hints at a state channel.